### DISCUSSION TITLE
How to set theme default to light?

### DISCUSSION THREAD

**Discussion by ganjarsetia** [kingyue737/vitify-next#3] [Q&A] [answered]
I tried to change [vuetify.ts](https://github.com/kingyue737/vitify-next/blob/79bcea96e259d3fff656d4339a474dda9ac5ee8c/src/plugins/vuetify.ts#L69) to `defaultTheme: 'light',` it wont change to light by default. I need to click toggle dark button to make it light.

So how to set theme default to light?

Nice template btw :+1:

**Comment by kingyue737** [comment, marked as answer]
Nice catch! `defaultTheme` in vuetify config will always be overridden by [`useDark`](https://vueuse.org/core/useDark/) which uses system preferred color mode by default (dark in your case). You can remove `defaultTheme` in vuetify.ts which does not take effect.

To set theme default to light on a system preferring dark, you can add following in [AppBar.vue](https://github.com/kingyue737/vitify-next/blob/main/src/components/layout/AppBar.vue):

```diff
const isDark = useDark({
+ writeDefaults: true,
+ initialValue: 'light',
  onChanged(dark: boolean) {
    theme.global.name.value = dark ? 'dark' : 'light'
  },
})
```

**Reply by kingyue737** [reply]
Also, you may need to clear localStorage to make the change take effect as `useDark` caches the theme in localStorage.

**Reply by ganjarsetia** [reply]
Thanks for the reply :smile: I read the docs about [useDark](https://vueuse.org/core/useDark/) even the source! I don't see about `initialValue`, but you know it. You really smart & knowledgeable man!

My alternative solution is

```diff
// file: src/App.vue
import { THEME_KEY } from 'vue-echarts'
+ import { useColorMode } from '@vueuse/core'

const theme = useTheme()
+ const mode = useColorMode()
+ mode.value = 'light'
```

I never clear the localStorage, just reopen in Incognito mode.

I don't know which is better.

If you think this is a bug, should I create new issue?

Greetings from ID, South East Asia :grin:

**Reply by kingyue737** [reply]
`UseDarkOptions` extends `UseColorModeOptions` which has `initialValue`
```ts
export interface UseDarkOptions
  extends Omit<UseColorModeOptions<BasicColorSchema>, "modes" | "onChanged"> {
```

Or you can trigger auto-complete in your IDE, TypeScript will tell you which options it has.

I think your solution is also good. I prefer setting `initialValue`, since `mode.value = 'light'` will force the site to be light on startup even if the user has already switched to dark once.
